### DISCUSSION TITLE
Is there a way to execute an entire file without moving the cursor?

### DISCUSSION THREAD

**Discussion by uga-rosa** [michaelb/sniprun#109] [Q&A]
If I use `:%SnipRun`, the cursor will be moved to the beginnnig of the file.

**Comment by michaelb** [comment]
By my beard, you're right. how the ???

**Comment by michaelb** [comment]
I was able to reproduce with a minimal setup and user-defined functions. 

By all evidence, this is not due to sniprun, but is a bug within vim/neovim ..? I'll submit a bug report.

If that can make your life easier, \<C-o> makes you jump to your last position

(edit: make the Ctrl+o workaround visible in markdown)

**Reply by uga-rosa** [reply]
It's an inconvenience at the beginning and at the end, so I hope it gets fixed :_(

**Reply by michaelb** [reply]
https://github.com/neovim/neovim/issues/16244

What a strange bug though. It's unlikely I'll be able to modify the part of the code that's causing it though, there isn't 40 ways of doing what i want to do

**Reply by michaelb** [reply]
You could maybe map a hotkey to 

%Sniprun + \<C-o>

? (Not in front of my PC anymore to figure out the right syntax)

**Reply by uga-rosa** [reply]
Unfortunately...
